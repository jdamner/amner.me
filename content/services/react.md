---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
