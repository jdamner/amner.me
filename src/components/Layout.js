import React, { Fragment } from "react"

import "@fontsource/comfortaa";
import "@fontsource/poiret-one";
import "@fontsource/limelight";

import Signpost from "./Signpost";

import "../sass/bootstrap.scss";
import "../sass/theme.scss";


export default function Layout({ children, title }) {
    const year = new Date().getFullYear();

    return (
        <Fragment>
            <title>{title + " - Amner.me"}</title>
            <main id="page" className="site">
                <Signpost />
                <div id='home' className='container'>
                    {children}
                </div>
            </main>
            <footer id="footer">
                <small>&copy; James Amner {year}</small>
            </footer>
        </Fragment>
    )
}