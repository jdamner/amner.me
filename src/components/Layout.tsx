/* API */
import { motion, useReducedMotion } from "framer-motion";
import { pageView } from "../api/Insights";

/* Components */
import Head from "next/head";
import ContactDetails from "./Global/ContactDetails";
import NavBar from "./Global/NavBar";
import Signpost from "./Global/Signpost";

/* Types */
import { children } from "../types/children.type";
import Link from "next/link";

export default function Layout({ children, title }: { children: children, title: string }) {
  //  Log page views
  pageView();

  const reducedMotion = useReducedMotion();

  return (
    <>
      <Head>
        <title>{title + " - Amner.me"}</title>
        <meta name="description" content={title} />
      </Head>
      <NavBar>
        <Signpost title='Contact'>
          <ContactDetails />
        </Signpost>
      </NavBar>
      <main id="page">
        <motion.div
          initial={{ opacity: reducedMotion ? 1 : 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: reducedMotion ? 1 : 0 }}
          transition={{ duration: 0.25 }}
        >
          {children}
        </motion.div>
      </main>
      <footer id="footer" className="container mx-auto px-3 md:px-0 my-3">
        <ul className="prose prose-ul:list-none max-w-full flex flex-col md:flex-row justify-between text text-xs">
          <li><Link href="/privacy">Privacy Policy</Link></li>
          <li>&copy; James Amner {new Date().getFullYear()}</li>
          <li><Link href="https://github.com/jdamner">GitHub</Link></li>
        </ul>
      </footer>
    </>
  )
}