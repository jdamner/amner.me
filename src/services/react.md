---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.