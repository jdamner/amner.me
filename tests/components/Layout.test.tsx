import React from "react"
import { render } from "@testing-library/react"
import Layout from "../../src/components/Layout"

describe("Layout", () => {
  it("renders correctly", () => {
    const { container } = render(<Layout title="TEST"><p>Empty</p></Layout>)
    expect(container).toMatchSnapshot()
  })
})
