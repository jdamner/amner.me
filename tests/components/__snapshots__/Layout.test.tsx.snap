// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <nav
    class="w-full h-16 bg-gradient-to-r bg-emerald-900 text-slate-100 from-teal-900 to-emerald-600"
  >
    <div
      class="container mx-auto h-full flex flex-row justify-between items-center px-3 md:px-0"
    >
      <ul>
        <li
          class="inline-block mr-3"
        >
          <a
            class="text-xl font-bold"
            href="/"
          >
            amner.me
          </a>
        </li>
      </ul>
      <button
        class="border-solid border-2 border-slate-800 uppercase font-black font-sans bg-white px-4 py-2 text-slate-800 hover:bg-slate-100 hover:text-black"
      >
        Contact
      </button>
    </div>
  </nav>
  <main
    id="page"
  >
    <div
      style="opacity: 0;"
    >
      <p>
        Empty
      </p>
    </div>
  </main>
  <footer
    class="container mx-auto text-center my-3"
    id="footer"
  >
    <small
      class="text-xs text"
    >
      Â© James Amner 
      2023
    </small>
  </footer>
</div>
`;
